core/src/OpenMP/Kokkos_OpenMP_Instance.cpp

```diff
@@ -338,17 +338,20 @@ void OpenMPInternal::initialize(int thread_count) {
   }
 
   // Check for over-subscription
+  auto const mpi_local_size = []() {
+    auto ranks = mpi_ranks_per_node();
+    if (ranks < 0) return 1;
+    return ranks;
+  }();
```

Comment:
```suggestion
  const auto reported_ranks = mpi_ranks_per_node();
  const auto mpi_local_size = reported_ranks < 0 ? 1 : reported_ranks;
```

This seems much less roundabout

If C++ were a more expression-oriented language, this would of course be less troublesome.

I'm just concerned about generated code size and representation, if the compiler isn't aggressive about inlining constructs like the lambda.

Reply:
> I'm just concerned about generated code size and representation, if the compiler isn't aggressive about inlining constructs like the lambda.

I am not convinced there is anything to worry about here.
https://godbolt.org/z/3c79KeKb8